test/hashmaps/test_maps.fypp

```diff
@@ -108,32 +111,54 @@ contains
         type(other_type) :: other
         logical :: conflict
 
-        do index2=1, size(test_8_bits), test_block
-            call set( key, test_8_bits( index2:index2+test_block-1 ) )
+        do index2=1, test_size, test_block
+            
             if (allocated(dummy)) deallocate(dummy)
-            dummy_val % value = test_8_bits( index2:index2+test_block-1 )
+            dummy_val % value = test_8_bits( index2:index2+test_block-1, 1 )
             allocate( dummy, source=dummy_val )
             call set ( other, dummy )
+            
+            ! Test all key interfaces
+            call set( key, test_8_bits( index2:index2+test_block-1, 1 ) )
+            call map % map_entry( key, other, conflict )
+            call check(error, .not.conflict, "Unable to map int8 entry because of a key conflict.")
+            
+            call set( key, transfer( test_8_bits( index2:index2+test_block-1, 2 ), [0_int32] ) )
             call map % map_entry( key, other, conflict )
-            call check(error, .not.conflict, "Unable to map entry because of a key conflict.")
+            call check(error, .not.conflict, "Unable to map int32 entry because of a key conflict.")
+
+            call set( key, to_string( transfer( test_8_bits( index2:index2+test_block-1, 3 ), 0_int32 ) ) )
+            call map % map_entry( key, other, conflict )
+            call check(error, .not.conflict, "Unable to map character entry because of a key conflict.")
+
             if (allocated(error)) return
+          
         end do
 
     end subroutine
 
     subroutine test_inquire_data(error, map, test_8_bits, test_block)
         type(error_type), allocatable, intent(out) :: error
         type(chaining_hashmap_type), intent(inout)  :: map
-        integer(int8), intent(in)               :: test_8_bits(test_size)
+        integer(int8), intent(in)               :: test_8_bits(test_size, key_types)
         integer(int_index), intent(in)          :: test_block
         integer :: index2
         logical :: present
         type(key_type) :: key
 
-        do index2=1, size(test_8_bits), test_block
-            call set( key, test_8_bits( index2:index2+test_block-1 ) )
+        do index2=1, test_size, test_block
+            call set( key, test_8_bits( index2:index2+test_block-1, 1 ) )
+            call map % key_test( key, present )
+            call check(error, present, "Int8 KEY not found in map KEY_TEST.")
+
+            call set( key, transfer( test_8_bits( index2:index2+test_block-1, 2 ), [0_int32] ) )
+            call map % key_test( key, present )
+            call check(error, present, "Int32 KEY not found in map KEY_TEST.")
+
+            call set( key, to_string( transfer( test_8_bits( index2:index2+test_block-1, 3 ), 0_int32 ) ) )
```

Comment:
Could you add a comment to explain the different tests (and their purposes)? It might help reviewers and future developers.

Reply:
I've added some additional comments per discussion above.   Feel free to add more.   I've also removed the character key type test.  The code worked fine on my computer, but failed the CI checks for reasons I don't understand.  It was a nice to have, but out of scope for this pull request which is for int32 key types.   

Also, per the note above, I think test_maps.fypp should be refactored to remove duplicate code prior to adding other key type checks to make code addition less error prone.

Reply:
> I've added some additional comments per discussion above.

Great. Thank you.

> I've also removed the character key type test. The code worked fine on my computer, but failed the CI checks for reasons I don't understand. It was a nice to have, but out of scope for this pull request which is for int32 key types.

I agree with you. Let's keep this PR focused on `int32`